classes/DataWarehouse/Visualization/AggregateChart.php

```diff
@@ -0,0 +1,1723 @@
+                    $this->_chart['layout']['barmode'] = 'group';
+                }
+                if ($data_description->combine_type=='stack') {
+                    $error_trace['y'] = array_fill(0, count($error_trace['y']), 0);
+                    for ($i = 0; $i < count($errorLabels); $i++) {
+                        if (!isset($errorLabels[$i])) {
+                            $error_trace['y'][$i] = null;
+                        }
+                    }
+                    $this->_chart['layout']['barmode'] = 'stack';
+                }
+            }
+
+            if (isset($trace['visible']) && $trace['visible'] != 1) {
+                $error_trace['visible'] = 'legendonly';
+            }
+
+            if(!$data_description->log_scale && $data_description->std_err)
+            {
+                if ($visible == 1) {
+                    $idx = count($this->_chart['data']) - 1;
+                    if ($this->_swapXY) {
+                        $this->_chart['data'][$idx]['error_x'] = $error_y;
+                    } else {
+                        $this->_chart['data'][$idx]['error_y'] = $error_y;
+                    }
+                }
+                $this->_chart['data'][] = $error_trace;
+            }
+
+            return array('data_labels' => $dataLabels, 'error_labels' => $errorLabels);
+        } // if not pie
+    } // end closure for buildErrorDataSeries function
+
+        /**
+         * Add a warning to the chart that data may be restricted.
+         */
+    protected function addRestrictedDataWarning() {
+        $roleRestrictionsStrings = $this->roleRestrictionsStringBuilder->getRoleRestrictionsStrings();
+        if (empty($roleRestrictionsStrings)) {
+                return;
+        }
+
+        $this->_chart['layout']['annotations'][] = array(
+            'name' => 'Restricted Data Warning',
+            'text' => implode('<br />', $roleRestrictionsStrings),
+            'xref' => 'paper',
+            'yref' => 'paper',
+            'xanchor' => 'left',
+            'yanchor' => 'bottom',
+            'bgcolor' => '#DFDFDF',
+            'showarrow' => false,
+            'x' => 0.0,
+            'y' => 1.0,
+        );
+    }
+
+    // ---------------------------------------------------------
+    // setChartTitleSubtitle()
+    //
+    // @param font_size
+    //
+    // ---------------------------------------------------------
+    protected function setChartTitleSubtitle($font_size)
+    {
+        // set title and subtitle for chart
+        if($this->_chart['layout']['annotations'][0]['text'] == '' && $this->_chart['layout']['annotations'][1]['text'] != '')
+        {
+            $this->setTitle($this->_chart['layout']['annotations'][1]['text'], $font_size);
+            $this->setSubtitle('', $font_size);
+        }
+    }
+
+    // ---------------------------------------------------------
+    // exportJsonStore()
+    //
+    // Export chart object aoverlayings JSON, indicate count.
```

Comment:
```suggestion
    // Export chart object as JSON, indicate count.
```
Looks like a paste typo (paste-o?)